### PR TITLE
Support to run batched inference in underlying changes

### PR DESCRIPTION
**Pull request by gc-fu**
This PR mainly modifies the following parts:

1. Scheduler
2. Underlying model

After apply this PR, we can run simple batched inference.  However, the output is still not correct and need further adjust.

### CODE REVIEW

File: vllm/core/scheduler.py [hunk 1]
```diff
@@ -391,3 +395,293 @@ def _swap_out(
...
+        self.scheduler_config = scheduler_config
+        #self.cache_config = cache_config
+
+        self.prompt_limit = min(self.scheduler_config.max_model_len,
+                                self.scheduler_config.max_num_batched_tokens)
+
+        # Instantiate the scheduling policy.
+        self.policy = PolicyFactory.get_policy(policy_name="fcfs")
+
+        # TODO(zhuohan): Use deque instead of list for better performance.
```
Comment (xiangyuT) [inline vllm/core/scheduler.py]: maybe confusing?

Comment (gc-fu) [reply vllm/core/scheduler.py]: deleted

File: vllm/engine/async_llm_engine.py [hunk 2]
```diff
@@ -309,10 +313,19 @@ async def engine_step(self) -> bool:
         """Kick the engine to process the waiting requests.
 
         Returns True if there are in-progress requests."""
-
+        print("One loop in engine_step")
+        # problem here is that we return blank requests
```
Comment (xiangyuT) [inline vllm/engine/async_llm_engine.py]: Does this problem still exist?

Comment (gc-fu) [reply vllm/engine/async_llm_engine.py]: no longer exist, will delete the print statements and the comment

**Review by xiangyuT — APPROVED** [review]
LGTM

**Pull request merged**